packages/big-design/src/components/Textarea/Textarea.tsx

```diff
@@ -21,88 +21,105 @@ interface PrivateProps {
 
 export type TextareaProps = Props & React.TextareaHTMLAttributes<HTMLTextAreaElement>;
 
-class StyleableTextarea extends React.PureComponent<TextareaProps & PrivateProps> {
-  static readonly defaultProps: Partial<Props> = {
-    rows: 3,
-    resize: true,
-  };
-
-  static Description = FormControlDescription;
-  static Error = FormControlError;
-  static Label = FormControlLabel;
-  private readonly uniqueId = uniqueId('textarea_');
-  private readonly MAX_ROWS = 7;
-
-  render() {
-    const { description, label, labelId, resize, rows, forwardedRef, ...props } = this.props;
-    const id = this.getId();
-
-    return (
-      <div>
-        {this.renderLabel()}
-        {this.renderDescription()}
-        <StyledTextareaWrapper>
-          <StyledTextarea {...props} id={id} rows={this.getRows(rows)} resize={resize} ref={forwardedRef} />
-        </StyledTextareaWrapper>
-      </div>
-    );
-  }
+const StyleableTextarea: React.FC<TextareaProps & PrivateProps> = ({
+  description,
+  error,
+  forwardedRef,
+  label,
+  labelId,
+  rows = 3,
+  resize = true,
+  ...props
+}) => {
+  const id = useMemo(() => (props.id ? props.id : uniqueId('textarea_')), [props.id]);
+  const MAX_ROWS = 7;
+
+  const renderedLabel = useMemo(() => {
+    if (typeof label === 'string') {
+      return (
+        <FormControlLabel id={labelId} htmlFor={id} renderOptional={!props.required}>
+          {label}
+        </FormControlLabel>
+      );
+    }
 
-  private getId() {
-    const { id } = this.props;
+    if (React.isValidElement(label) && label.type === FormControlLabel) {
+      return React.cloneElement(label as React.ReactElement<React.LabelHTMLAttributes<HTMLLabelElement>>, {
+        id: labelId,
+        htmlFor: id,
+      });
+    }
 
-    return id ? id : this.uniqueId;
-  }
+    if (!label) {
```

Comment:
What do you think about movin this check to be the first one? Corner case: `label = ''`.